tractor/_actor.py

```diff
@@ -575,6 +576,17 @@ async def _async_main(
                         await chan.connect()
                         # initial handshake, report who we are, who they are
                         await self._do_handshake(chan)
+
+                        if self._spawn_method == "trio":
+                            # recieve additional init params
+                            parent_data = await chan.recv()
+                            for attr, value in parent_data.items():
+                                setattr(self, attr, value)
+
+                            # update local arbiter_addr var
+                            if "_arb_addr" in parent_data:
```

Comment:
Is there any reason this one is set separately from the loop above?

Reply:
Yes, because its not updating ``self`` attributes at all. Its updating a local variable needed in the finally block of the ``_async_main``.